marklogic-data-hub/src/main/java/com/marklogic/hub/impl/ArtifactManagerImpl.java

```diff
@@ -70,16 +76,28 @@ public ObjectNode getArtifactSettings(String artifactType, String artifactName)
     }
 
     public ObjectNode updateArtifactSettings(String artifactType, String artifactName, JsonNode settings) {
-        return (ObjectNode) getArtifactService().setArtifactSettings(artifactType, artifactName, settings);
+        ObjectNode resp = (ObjectNode) getArtifactService().setArtifactSettings(artifactType, artifactName, settings);
+        writeArtifactInProject(artifactType, (ObjectNode) settings, true);
+        return resp;
     }
 
     protected ArtifactService getArtifactService() {
         DatabaseClient dataServicesClient = hubConfig.newStagingClient(null);
         return ArtifactService.on(dataServicesClient);
     }
 
-    protected void writeArtifactInProject(String artifactType, ObjectNode artifact) {
-        Path fileLocation = buildArtifactProjectLocation(artifactType, getNameFromArtifact(artifactType, artifact), getVersionFromArtifact(artifactType, artifact));
+    protected void writeArtifactInProject(String artifactType, ObjectNode artifact, boolean isSetting) {
+        Path fileLocation;
+        if (!isSetting) {
+            fileLocation = buildArtifactProjectLocation(artifactType, getNameFromArtifact(artifactType, artifact), getVersionFromArtifact(artifactType, artifact), false);
+        } else {
+            if (!artifact.has("artifactName") || artifact.get("artifactName").asText().length() == 0) {
+                logger.error("Failed to write to project folder due to invalid settings! artifactType: (%s), artifact settings: (%s).", artifactType, artifact.toString());
```

Comment:
Since we assume that artifactName is not null, just let an exception propagate when "asText()" is called. If we want something nicer, we can check to see if "artifactName" doesn't exist and then do something like this:

    throw new RuntimeException("Unable to write artifact to project folder; no artifactName found")